llvm/test/Transforms/LoopUnroll/AArch64/vector.ll

```diff
@@ -485,12 +485,289 @@ exit:                                 ; preds = %vector.body
+; CORTEXA55:       [[EXIT]]:
+; CORTEXA55-NEXT:    ret void
+;
+entry:
+  %min.iters.check = icmp ult i64 %N, 8
+  br i1 %min.iters.check, label %for.body, label %vector.main.loop.iter.check
+
+vector.main.loop.iter.check:
+  %min.iters.check7 = icmp ult i64 %N, 32
+  br i1 %min.iters.check7, label %vec.epilog.ph, label %vector.ph
+
+vector.ph:
+  %n.vec = and i64 %N, -32
+  %broadcast.splatinsert = insertelement <16 x i8> poison, i8 %val, i64 0
+  %broadcast.splat = shufflevector <16 x i8> %broadcast.splatinsert, <16 x i8> poison, <16 x i32> zeroinitializer
+  br label %vector.body
+
+vector.body:
+  %index = phi i64 [ 0, %vector.ph ], [ %index.next, %vector.body ]
+  %0 = getelementptr inbounds nuw i8, ptr %p, i64 %index
+  %1 = getelementptr inbounds nuw i8, ptr %0, i64 16
+  %wide.load = load <16 x i8>, ptr %0, align 1
+  %wide.load8 = load <16 x i8>, ptr %1, align 1
+  %2 = add <16 x i8> %wide.load, %broadcast.splat
+  %3 = add <16 x i8> %wide.load8, %broadcast.splat
+  store <16 x i8> %2, ptr %0, align 1
+  store <16 x i8> %3, ptr %1, align 1
+  %index.next = add nuw i64 %index, 32
+  %4 = icmp eq i64 %index.next, %n.vec
+  br i1 %4, label %middle.block, label %vector.body, !llvm.loop !2
+
+middle.block:
+  %cmp.n = icmp eq i64 %N, %n.vec
+  br i1 %cmp.n, label %exit, label %vec.epilog.iter.check
+
+vec.epilog.iter.check:
+  %n.vec.remaining = and i64 %N, 24
+  %min.epilog.iters.check = icmp eq i64 %n.vec.remaining, 0
+  br i1 %min.epilog.iters.check, label %for.body, label %vec.epilog.ph
+
+vec.epilog.ph:
+  %vec.epilog.resume.val = phi i64 [ %n.vec, %vec.epilog.iter.check ], [ 0, %vector.main.loop.iter.check ]
+  %n.vec10 = and i64 %N, -8
+  %broadcast.splatinsert11 = insertelement <8 x i8> poison, i8 %val, i64 0
+  %broadcast.splat12 = shufflevector <8 x i8> %broadcast.splatinsert11, <8 x i8> poison, <8 x i32> zeroinitializer
+  br label %vec.epilog.vector.body
+
+vec.epilog.vector.body:
+  %index13 = phi i64 [ %vec.epilog.resume.val, %vec.epilog.ph ], [ %index.next15, %vec.epilog.vector.body ]
+  %5 = getelementptr inbounds nuw i8, ptr %p, i64 %index13
+  %wide.load14 = load <8 x i8>, ptr %5, align 1
+  %6 = add <8 x i8> %wide.load14, %broadcast.splat12
+  store <8 x i8> %6, ptr %5, align 1
+  %index.next15 = add nuw i64 %index13, 8
+  %7 = icmp eq i64 %index.next15, %n.vec10
+  br i1 %7, label %vec.epilog.middle.block, label %vec.epilog.vector.body, !llvm.loop !3
+
+vec.epilog.middle.block:
+  %cmp.n16 = icmp eq i64 %N, %n.vec10
+  br i1 %cmp.n16, label %exit, label %for.body
+
+for.body:
+  %i.06 = phi i64 [ %inc, %for.body ], [ %n.vec10, %vec.epilog.middle.block ], [ %n.vec, %vec.epilog.iter.check ], [ 0, %entry ]
+  %arrayidx = getelementptr inbounds nuw i8, ptr %p, i64 %i.06
+  %8 = load i8, ptr %arrayidx, align 1
+  %add = add i8 %8, %val
+  store i8 %add, ptr %arrayidx, align 1
+  %inc = add nuw i64 %i.06, 1
+  %exitcond.not = icmp eq i64 %inc, %N
+  br i1 %exitcond.not, label %exit, label %for.body, !llvm.loop !4
+
```

Comment:
Can this test be simplified? The main bit is the scalar loop with the metadata and scalar instructions?